src/plugins/intel_gpu/src/graph/fully_connected.cpp

```diff
@@ -117,7 +117,7 @@ layout fully_connected_inst::calc_output_layout(fully_connected_node const& node
         feature = std::max({input_layout.spatial(0), input_layout.spatial(1), input_layout.spatial(2)});
     }
 
-    if (desc->input_size > 3) {
+    if (desc->input_size > 4) {
        input_layout.set_partial_shape(reshape_to_2d(input_pshape, feature));
```

Comment:
If shape is 5d, e.g.,
In [1,3,5,10,4] W [1,3,5,4,20]
feature is detected as inputshape[3] => 10. 
However actually it is expected to be recognized as 4. 
Seems this is not still covered by the test.

Reply:
Yes, this PR was extended fc 3d case to support fc 4d case.